pydicom/tests/test_data_manager.py

```diff
@@ -255,6 +256,13 @@ def test_get_testdata_files_external_ignore_hash(self):
 @pytest.mark.skipif(EXT_PYDICOM, reason="pydicom-data installed")
 class TestDownload:
     """Tests for the download module."""
+
+    def test_get_testdata_file_no_download(self, recwarn):
+        """Test that external data source preferred over cache."""
```

Comment:
Which cache does this refer to?

Reply:
The one referred by the `data_path_with_download` function:

> Return the absolute path to the cached file with filename.
> If the file isn't available in the cache then it will be downloaded.

The `download` flag switches on/off the call to this function.